src/Microsoft.AspNetCore.Mvc.Core/Internal/MvcCoreLoggerExtensions.cs

```diff
@@ -417,35 +438,141 @@ static MvcCoreLoggerExtensions()
+
+            _doneAttemptingToBindModel = LoggerMessage.Define<Type, string>(
+                LogLevel.Debug,
+                25,
+                "Done attempting to bind model of type '{ModelType}' using the name '{ModelName}'.");
+
+            _attemptingToValidateParameter = LoggerMessage.Define<string, string, Type>(
+                LogLevel.Debug,
+                26,
+                "Attempting to validate the bound {ParameterOrProperty} '{ParameterOrPropertyName}' of type '{ModelType}' ...");
+
+            _doneAttemptingToValidateParameter = LoggerMessage.Define<string, string, Type>(
+                LogLevel.Debug,
+                27,
+                "Done attempting to validate the bound {ParameterOrProperty} '{ParameterOrPropertyName}' of type '{ModelType}'.");
+
+            _noNonIndexBasedFormatFoundForCollection = LoggerMessage.Define<string, string>(
+                LogLevel.Debug,
+                28,
+                "Could not bind to collection using a format like {ModelName}=value1&{ModelName}=value2");
+
+            _attemptingToBindCollectionUsingIndices = LoggerMessage.Define<string, string, string, string, string, string>(
+                LogLevel.Debug,
+                29,
+                "Attempting to bind model using indices. Example formats include: " +
+                "[0]=value1&[1]=value2, " +
+                "{ModelName}[0]=value1&{ModelName}[1]=value2, " +
+                "{ModelName}.index=zero&{ModelName}.index=one&{ModelName}[zero]=value1&{ModelName}[one]=value2");
+
+            _attemptingToBindCollectionOfKeyValuePair = LoggerMessage.Define<string, string, string, string, string, string>(
+                LogLevel.Debug,
+                30,
+                "Attempting to bind collection of KeyValuePair. Example formats include: " +
+                "[0].Key=key1&[0].Value=value1&[1].Key=key2&[1].Value=value2, " +
+                "{ModelName}[0].Key=key1&{ModelName}[0].Value=value1&{ModelName}[1].Key=key2&{ModelName}[1].Value=value2, " +
+                "{ModelName}[key1]=value1&{ModelName}[key2]=value2");
+
+            _noKeyValueFormatForDictionaryModelBinder = LoggerMessage.Define<string, string, string>(
+                LogLevel.Debug,
+                33,
+                "Attempting to bind model with name '{ModelName}' using the format {ModelName}[key1]=value1&{ModelName}[key2]=value2");
+
+            _ifMatchPreconditionFailed = LoggerMessage.Define<EntityTagHeaderValue>(
+                LogLevel.Debug,
+                34,
                 "Current request's If-Match header check failed as the file's current etag '{CurrentETag}' does not match with any of the supplied etags.");
 
             _ifUnmodifiedSincePreconditionFailed = LoggerMessage.Define<DateTimeOffset?, DateTimeOffset?>(
                 LogLevel.Debug,
-                13,
+                35,
```

Comment:
As @Eilon mentioned above, don't renumber existing log messages. Start at the end with the new messages.

Reply:
The above log message `_ifUnmodifiedSincePreconditionFailed ` was added as part of preview1, so we are ok changing it here.